contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -1032,11 +1081,11 @@ private void getAndMapCompoundData(String path, List<String> fieldNames, IHDF5Re
 
           SchemaBuilder innerSchema = new SchemaBuilder();
           MapBuilder mapBuilder = innerSchema.addMap(COMPOUND_DATA_FIELD_NAME);
-
           for (HDF5CompoundMemberInformation info : infos) {
             fieldNames.add(info.getName());
+            String compoundColumnDataType = info.getType().tryGetJavaType().getSimpleName();
 
-            switch (info.getType().tryGetJavaType().getSimpleName()) {
+            switch (compoundColumnDataType) {
```

Comment:
Would be more reliable to switch on the type itself. Switch probably won't work, so would instead need a chain of `if` statements.

Reply:
The `getType()` is actually a poorly named method in that it doesn't return the type, but rather an `HDF5DataTypeInformation` object. [1]. Once you have this object, you still have to call one of the various methods like `tryGetJavaType()` to actually get a data type that is useful. 

[1]: http://svnsis.ethz.ch/doc/openbis/S175.0/ch/systemsx/cisd/hdf5/HDF5DataTypeInformation.html

Reply:
If you think that's better I can change it.

Reply:
Ah, more complex than a quick glance at the code suggested. OK, let's leave it.